### ISSUE TITLE
IsRemoteReachable(Uri uri, TimeSpan timeout) - timeout parameter is ignored on Android device

### ISSUE THREAD

**Issue by abrasat** [jamesmontemagno/ConnectivityPlugin#117]
I am testing if a server is reachable in Internet by using this call:

```
var  inetConnOk = await currConnectivity.IsRemoteReachable(serverUri, new TimeSpan(0, 0, 3));
```

On an Android tablet which does not have Internet access, it takes more than 10 seconds until the function returns the "false" value, instead of returning after the max. 3 seconds set as timeout.

**Comment by jamesmontemagno** [comment]
If you are creating an issue for a BUG please fill out this information. If you are asking a question or requesting a feature you can delete the sections below. 

**Failure to fill out this information will result in this issue being closed.** If you post a full stack trace in a bug it will be closed, please post it to http://gist.github.com and then post the link here.

## Bug Information

Version Number of Plugin:
Device Tested On:
Simulator Tested On:
Version of VS: 
Version of Xamarin:
Versions of other things you are using:

### Steps to reproduce the Behavior

### Expected Behavior

### Actual Behavior

### Code snippet

### Screenshots

**Comment by abrasat** [comment]
Bug Information

Version Number of Plugin: 4.0.0-beta139
Device Tested On: Asus Zen Pad
Simulator Tested On:
Version of VS: Professional 2017, Version 15.4.3
Version of Xamarin:  Xamarin.Forms 2.4.0.74863, Xamarin 4.7.10.38, Xamarin.Android SDK 8.0.2.1
Versions of other things you are using:
Steps to reproduce the Behavior
- Android Device connected to router via WiFi. Router not connected to Internet.
- Application checks internet availability by calling IsRemoteReachable() with server uri and timeout set to 3 seconds.

Expected Behavior
IsRemoteReachable() returns false after 3 seconds.

Actual Behavior
IsRemoteReachable() returns false after about 8 seconds.

Code snippet
```
var watch = System.Diagnostics.Stopwatch.StartNew();
var inetConnOk = await currConnectivity.IsRemoteReachable(srvUri, new TimeSpan(0, 0, 3));
watch.Stop();
var elapsedMs = watch.ElapsedMilliseconds;
logger.LogInformation($"CheckInetConn {elapsedMs}, inetConnStatus {inetConnOk}");
```
Screenshots

**Comment by abrasat** [comment]
Bug Information

Version Number of Plugin: 4.0.0-beta139
Device Tested On: 
Simulator Tested On: iPhone Simulator on MacBook Pro
Version of VS: Professional 2017, Version 15.4.3
Version of Xamarin:  Xamarin.Forms 2.4.0.74863, Xamarin 4.7.10.38, Xamarin.iOS SDK 11.3.0.47
Versions of other things you are using:
Steps to reproduce the Behavior
- MacBook connected to router via WiFi. Router not connected to Internet.
- Application checks internet availability by calling IsRemoteReachable() with server uri and timeout set to 3 seconds.

Expected Behavior
IsRemoteReachable() returns false after 3 seconds.

Actual Behavior
IsRemoteReachable() returns false after about 33 seconds.

Code snippet
```
var watch = System.Diagnostics.Stopwatch.StartNew();
var inetConnOk = await currConnectivity.IsRemoteReachable(srvUri, new TimeSpan(0, 0, 3));
watch.Stop();
var elapsedMs = watch.ElapsedMilliseconds;
logger.LogInformation($"CheckInetConn {elapsedMs}, inetConnStatus {inetConnOk}");
```
Screenshots

**Comment by abrasat** [comment]
Any news regarding above bugs ? Are there any fixes planned ?

**Comment by jamesmontemagno** [comment]
We prioritize and work on bugs as necessary and do not offer any timelines.

**Comment by OnkelMato** [comment]
I have the same issue to check if a REST service endpoint is available or not. I was training to set the timeout on `httpclient` and after googling to the xamarin internal settings in `AppDelegate.cs`. See [here](https://coderedirect.com/questions/241277/xamarin-ios-httpclient-timeout-doesnt-work).

Finally, I tried this library but, this does not work too. I tried both functions, `IsReachable` and `IsRemoteReachable`, whereby IsReachable never returned and IsRemoteReachable has an unchanged timeout (2 mins in my case). In my case, the host does not exist (command line ping not working too).

Hope this helps.
